sql/api/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -417,7 +417,7 @@ case class StructType(fields: Array[StructField]) extends DataType with Seq[Stru
   override def defaultSize: Int = fields.map(_.dataType.defaultSize).sum
 
   override def simpleString: String = {
-    val fieldTypes = fields.view.map(field => s"${field.name}:${field.dataType.simpleString}").toSeq
+    val fieldTypes = fields.map(field => s"${field.name}:${field.dataType.simpleString}").toSeq
```

Comment:
No， The `.toSeq` will trigger computation, so this view doesn't achieve its intended effect, unless we refactor the truncatedString method to accept SeqView instead of Seq.

Reply:
If we want to do this refactor for `SparkStringUtils#truncatedString` function, I can revert this change and create a Jira to track it."

Reply:
I guess the `view` was added here first for performance consideration. Later, the `toSeq` was added for compatibility with Scala 2.13, breaking the original intention.

Reply:
Got, let me test to use `LazyList` here